notebook/static/notebook/js/notebook.js

```diff
@@ -2876,10 +2876,12 @@ define([
                                 cell.output_area.trusted = true;
                             }
                         }
-                        nb.events.on('notebook_saved.Notebook', function () {
-                            window.location.reload();
+                        nb.contents.trust(nb.notebook_path)
+                            .then(function(res) {
+                                nb.events.trigger("trust_changed.Notebook", true)
```

Comment:
it should still trigger reload to get the newly trusted notebook without sanitized output.

And `.save` should only be avoided when the notebook is read-only. We should still save & reload for non-read-only notebooks to avoid losing unsaved changes (which is unavoidable for read-only).